air/src/execution_step/boxed_value/stream_map.rs

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use std::rc::Rc;
+
+use super::ValueAggregate;
+use crate::execution_step::ExecutionResult;
+use crate::JValue;
+
+use super::stream::*;
+
+use air_interpreter_data::GenerationIdx;
+use air_trace_handler::merger::ValueSource;
+use air_trace_handler::TraceHandler;
+use serde::Serialize;
+use serde_json::json;
+
+#[derive(Debug, Default, Clone)]
+pub struct StreamMap {
+    stream: Stream,
+}
+
+impl StreamMap {
+    fn envelope(key: &(impl Into<JValue> + Serialize), value: Rc<JValue>) -> Rc<JValue> {
```

Comment:
Rc is useless here, as `serde_json::Value` can't hold Rc, and `json!` will just clone the inner value.